device_samples/getting_started/src/receive_messages/receive_message.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+// Choose a protocol by uncommenting one of these transports.
+const Protocol = require('azure-iot-device-mqtt').Mqtt;
+// var Protocol = require('azure-iot-device-amqp').Amqp;
+// var Protocol = require('azure-iot-device-http').Http;
+// var Protocol = require('azure-iot-device-mqtt').MqttWs;
+// var Protocol = require('azure-iot-device-amqp').AmqpWs;
+
+require('es5-shim');
+const Client = require('azure-iot-device').Client;
+let client = null;
+
+function main() {
+    // open a connection to the device
+    const deviceConnectionString = process.env.IOTHUB_DEVICE_CONNECTION_STRING || '';
+    
+    // make sure we have a connection string before we can continue
+    if (deviceConnectionString === null || deviceConnectionString === undefined) {
+        console.error('\x1b[31m%s\x1b[0m', 'Missing device connection string');
+        process.exit(0);
+    }
+    
+    client = Client.fromConnectionString(deviceConnectionString, Protocol);
+    client.open(onConnect);
+}
+
+function onConnect(err) {
+    if(!!err) {
+        console.error('\x1b[31m%s\x1b[0m', 'Could not connect: ' + err.message);
+    } else {
+        console.log('Connected to device. Registering message handler.');
+        client.on('message', handleMessage);        
+        console.log('Ready to recieve messages...');
+    }
+}
+
+function handleMessage(msg) {
+    console.log('\x1b[33m%s\x1b[0m', `\nReceived C2D message:`);
+    console.log(` message: ${msg.data.toString().trim()}`);
+    console.log(` properties: ${JSON.stringify(msg.properties.propertyList)}`);   
+}
+
```

Comment:
One issue here is that for Amqp and Http the user is supposed to "complete" the C2D message and for Mqtt is not.
Maybe check the transport constructor name and complete the message if appropriate?